sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StatefulProcessorHandleImpl.scala

```diff
@@ -313,3 +290,176 @@ class StatefulProcessorHandleImpl(
     }
   }
 }
+
+/**
+ * This DriverStatefulProcessorHandleImpl is used within TransformWithExec
+ * on the driver side to collect the columnFamilySchemas before any processing is
+ * actually done. We need this class because we can only collect the schemas after
+ * the StatefulProcessor is initialized.
+ */
+class DriverStatefulProcessorHandleImpl(timeMode: TimeMode)
+  extends StatefulProcessorHandleImplBase(timeMode) {
+
+  private[sql] val columnFamilySchemaUtils = ColumnFamilySchemaUtilsV1
+
+  // Because this is only happening on the driver side, there is only
+  // one task modifying and accessing this map at a time
+  private[sql] val columnFamilySchemas: mutable.Map[String, ColumnFamilySchema] =
+    new mutable.HashMap[String, ColumnFamilySchema]()
+
+  def getColumnFamilySchemas: Map[String, ColumnFamilySchema] = columnFamilySchemas.toMap
+
+  /**
```

Comment:
Btw, I'm not sure users will try to do something except assigning the returning value to the field, but if they are ever doing something, this will lead to NPE.

Probably need to document this in the method doc of base class? cc. @anishshri-db